[docs][PLAT-10799] Additional steps to clarify certificate verification for encryption in transit (#19473)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-platform/security/enable-encryption-in-transit.md

#### Verify certificate chain

Perform the following steps to verify your certificates:

> 1. Execute the following verify command which checks the database node certificate "node.crt" against the root CA certificate "ca.crt":

Reviewer: ```suggestion
1. Execute the following verify command which checks the database node certificate (node.crt) against the root CA certificate (ca.crt):
```

    ```sh

>     openssl verify -CAfile bundle.pem cert.pem

Reviewer 2: minor: Given we refer to this as ca.crt and node.crt above, we could use the same naming here.

    ```

    The `bundle.pem` file is a certificate bundle containing the root certificate and any intermediate certificates in the PEM format.

> 1. Verify that the node certificate (`node.ip_address.crt`) and the node private key (`node.ip_address.key`) match. See [How do I verify that a private key matches a certificate?](https://www.ssl247.com/knowledge-base/detail/how-do-i-verify-that-a-private-key-matches-a-certificate-openssl-1527076112539/ka03l0000015hscaay/)

Reviewer 2: minor: node.crt and node.key would be consistent

> 1. Verify that the node certificate expiration is at least 3 months by checking the validity field in the output of the following command:

Reviewer 2: We could ask them to verify this expiration date for the ca.crt as well

    ```sh
    openssl x509 -in node.crt -text -noout
    ```

> 1. Verify that the node certificate Common Name or Subject Alternate Name contains the IP address or DNS name of each on-prem node on which the nodes are deployed.

Reviewer 2: It would be good to clarify one issue here that customers run into -

the CN/SAN needs to have the same entry as the node's entry in the onprem provider. If the node was added as a DNS address in the onprem provider, the same DNS entry should be present here. 

I'm not sure how to convey this point exactly. Customers sometimes enter a DNS address in the onprem provider but add a cert with the resolved IP address.

>     If you face any issue with the above verification, you can customize the verification. Refer to [Customizing the verification of RPC server certificate by the client](https://www.yugabyte.com/blog/yugabytedb-server-to-server-encryption/#customizing-the-verification-of-rpc-server-certificate-by-the-client).

Reviewer 2: ... you can customize the level of certificate validation while creating a universe that uses these certificates.

#### Verify certificate chain

Perform the following steps to verify your certificates:

1. Execute the following verify command which checks the database node certificate (node.crt) against the root CA certificate (ca.crt):

    ```sh
    openssl verify ca.crt bundle.pem node.crt
    ```

    The `bundle.pem` file is a certificate bundle containing the root certificate and any intermediate certificates in the PEM format.

1. Verify that the node certificate (`node.crt`) and the node private key (`node.key`) match. See [How do I verify that a private key matches a certificate?](https://www.ssl247.com/knowledge-base/detail/how-do-i-verify-that-a-private-key-matches-a-certificate-openssl-1527076112539/ka03l0000015hscaay/)

1. Verify that the node certificate and Root CA certificate expiration is at least 3 months by checking the validity field in the output of the following commands:

    ```sh
    openssl x509 -in node.crt -text -noout
    ```

    ```sh
    openssl x509 -in root.crt -text -noout
    ```

1. Verify that the node certificate Common Name (CN) or Subject Alternate Name  (SAN) contains the IP address or DNS name of each on-prem node on which the nodes are deployed.

>     **Note** that the entry you provide for the CN or SAN should be the _same_ as the on-prem node. If the node is added as a DNS address in the on-prem provider, the same DNS entry should be present as the CN/SAN.

Reviewer: ```suggestion
    Each entry you provide for the CN or SAN must match the on-prem node as entered in the provider configuration. For example, if the node address is entered as a DNS address in the on-prem provider configuration, you must use the same DNS entry in the CN or SAN, not the resolved IP address.
```

Reviewer: Maybe warrants an actual note shortcode

    If you face any issue with the above verification, you can customize the level of certificate validation while creating a universe that uses these certificates. Refer to [Customizing the verification of RPC server certificate by the client](https://www.yugabyte.com/blog/yugabytedb-server-to-server-encryption/#customizing-the-verification-of-rpc-server-certificate-by-the-client).

{{< note >}}

> Note that the client certificates and keys are required only if you intend to use [PostgreSQL certificate based authentication](https://www.postgresql.org/docs/current/auth-pg-hba-conf.html#:~:text=independent%20authentication%20option-,clientcert,-%2C%20which%20can%20be).

Reviewer: ```suggestion
The client certificates and keys are required only if you intend to use [PostgreSQL certificate-based authentication](https://www.postgresql.org/docs/current/auth-pg-hba-conf.html#:~:text=independent%20authentication%20option-,clientcert,-%2C%20which%20can%20be).
```

#### Verify certificate chain

Perform the following steps to verify your certificates:

1. Execute the following verify command which checks the database node certificate (node.crt) against the root CA certificate (ca.crt):

    ```sh

>     openssl verify ca.crt bundle.pem node.crt

Reviewer 2: there should be no bundle.pem in here now

    ```

    The `bundle.pem` file is a certificate bundle containing the root certificate and any intermediate certificates in the PEM format.

1. Verify that the node certificate (`node.crt`) and the node private key (`node.key`) match. See [How do I verify that a private key matches a certificate?](https://www.ssl247.com/knowledge-base/detail/how-do-i-verify-that-a-private-key-matches-a-certificate-openssl-1527076112539/ka03l0000015hscaay/)

1. Verify that the node certificate and Root CA certificate expiration is at least 3 months by checking the validity field in the output of the following commands:

    ```sh
    openssl x509 -in node.crt -text -noout
    ```

    ```sh

>     openssl x509 -in root.crt -text -noout

Reviewer 2: We should use ca.crt here to be consistent
